Source/Controllers/MainViewControllers/TableStructure/TableSortHelper.swift

```diff
@@ -0,0 +1,117 @@
+//
+// Created by Luis Aguiniga on January 9, 2022.
+// Copyright (c) 2022 Sequel-Ace. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person
+//  obtaining a copy of this software and associated documentation
+//  files (the "Software"), to deal in the Software without
+//  restriction, including without limitation the rights to use,
+//  copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the
+//  Software is furnished to do so, subject to the following
+//  conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+//  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+//  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+//  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+//  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+//  OTHER DEALINGS IN THE SOFTWARE.
+//
+//  More info at <https://github.com/sequelpro/sequelpro>
+
+import AppKit
+
+/// Helper class to maintain the sort state changes of a Table View.
+/// - Important: Sorting with `NSSortDescriptors` uses `Key-Value Coding` so the data
+///   field names must match the `descriptor.key` as well as the `column.identifier`.
+@objc class TableSortHelper: NSObject {
+    let descriptors: [NSSortDescriptor]
+    let tableView: NSTableView
+    private(set) var currentColumn: NSTableColumn?
+    private(set) var currentOrder: SortOrder = .default
+
+    enum SortOrder { case ascending, descending, `default` }
+
+    @objc init(tableView: NSTableView, andDescriptors descriptors: [NSSortDescriptor]) {
```

Comment:
can we use naming without `and`?
`@objc init(tableView: NSTableView, descriptors: [NSSortDescriptor]) {`